supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/HtCondorJobExecutionActor.scala

```diff
@@ -93,14 +97,29 @@ class HtCondorJobExecutionActor(override val jobDescriptor: BackendJobDescriptor
     case ExecutionResultStored(hash) => log.debug("{} Cache entry was stored for Job with hash {}.", tag, hash)
 
     case ExecutionResultAlreadyExist => log.warning("{} Cache entry for hash {} already exist.", tag, jobHash)
+
+    case KvPair(key, Some(jobId)) if key.key == HtCondorJobIdKey =>
+      log.debug("{} Found job id {} for job {} trying to recover job now.", tag, jobId, jobDescriptor.key)
+      executionResponse success trackTaskToCompletion(jobId)
+
+    case KvKeyLookupFailed(_) =>
+      log.debug("{} Job id not found for job {} falling back to execute.", tag, jobDescriptor.key)
+      execute
+
+    case KvFailure(_, e) =>
+      val errMsg = s"$tag Failure attempting to look up HtCondor job id for key ${jobDescriptor.key}. Exception message: ${e.getMessage}."
+      log.error(errMsg)
+      executionResponse.tryFailure(e)
```

Comment:
Do we want to deviate from the earlier default behavior for recover? It used to execute if the recover functionality was not supported. I think we could stick to it and execute here rather than failing altogether (because of some other service failure).